tests/unit/test_operator.py

```diff
@@ -1,68 +1,161 @@
-import pytest
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+
+"""Unit tests for Seldon Core Operator/Charm."""
+
 import json
-from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from unittest.mock import MagicMock, patch
+
+import pytest
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
 from ops.testing import Harness
 
-from charm import Operator
+from charm import SeldonCoreOperator
+
 
+@pytest.fixture(scope="function")
+def harness() -> Harness:
+    """Create and return Harness for testing."""
+    return Harness(SeldonCoreOperator)
 
-@pytest.fixture
-def harness():
-    return Harness(Operator)
 
+class TestCharm:
+    """Test class for SeldonCoreOperator."""
 
-def test_not_leader(harness):
-    harness.begin_with_initial_hooks()
-    assert harness.charm.model.unit.status == WaitingStatus("Waiting for leadership")
+    @patch("charm.KubernetesServicePatch", lambda x, y, service_name: None)
+    def test_not_leader(self, harness: Harness):
+        """Test not a leader scenario."""
+        # setup container netwroking simulation
+        harness.set_can_connect("seldon-core", True)
+        harness.container_pebble_ready("seldon-core")
 
+        harness.begin_with_initial_hooks()
+        assert harness.charm.model.unit.status == WaitingStatus("Waiting for leadership")
 
-def test_missing_image(harness):
-    harness.set_leader(True)
-    harness.begin_with_initial_hooks()
-    assert harness.charm.model.unit.status == BlockedStatus(
-        "Missing resource: oci-image"
-    )
+    @patch("charm.KubernetesServicePatch", lambda x, y, service_name: None)
+    @patch("charm.SeldonCoreOperator.k8s_resource_handler")
+    @patch("charm.SeldonCoreOperator.configmap_resource_handler")
+    @patch("charm.SeldonCoreOperator.crd_resource_handler")
+    def test_no_relation(
+        self,
+        _: MagicMock,  # k8s_resource_handler
+        __: MagicMock,  # configmap_resource_handler
+        ___: MagicMock,  # crd_resource_handler
+        harness: Harness,
+    ):
+        """Test no relation scenario."""
+        harness.set_leader(True)
+        harness.add_oci_resource(
+            "oci-image",
+            {
+                "registrypath": "ci-test",
+                "username": "",
+                "password": "",
+            },
+        )
 
+        # setup container netwroking simulation
+        harness.set_can_connect("seldon-core", True)
+        harness.container_pebble_ready("seldon-core")
```

Comment:
Just a suggestion.  Nearly everywhere applies this to the harness produced by the harness fixture.  You could move this to the fixture.  Could also move `.set_leader(True)` and `.set_model_name(...)`

Reply:
`set_leader()` is used in other tests that test for not leader. So, I guess that one would stay.